server/routes/api/contest.js

```diff
@@ -138,20 +148,86 @@ contestRouter.post("/:id/submission", auth, (req, res) => {
     contest_id,
     user_id,
     upload_files,
-    user_name
+    user_name,
   });
 
   newSubmission
     .save()
     .then(() => {
-      res.json("Submission added successfully");
+      res.status(201).json("Submission added successfully");
     })
     .catch((err) => {
-      console.log(err.message)
-      res.status(400).json("Error: " + err.message);
+      console.log(err.message);
+      res.status(500).json("Error: " + err.message);
     });
 });
 
+// Route: PUT api/contest/:id/submission
+// Desc: Update a submission
+// access: private
+
+contestRouter.put("/:id/submission/:submission_id", auth, (req, res) => {
+  const { winner } = req.body;
+  console.log(req.body);
+  Contest.findById(req.params.id)
+    .then((contest) => {
+      if (contest) {
+        Submission.findById(req.params.submission_id)
+          .then((submission) => {
+            if (submission && contest.user_id === req.user.id) {
+              Submission.findByIdAndUpdate(req.params.submission_id, {
+                winner,
+              })
+                .then((r) => {
+                  User.findById(submission.user_id)
+                    .then((user) => {
+                      User.findByIdAndUpdate(submission.user_id, {
```

Comment:
ooo be careful with all these nested callbacks - you're going to run into a "callback hell" going down this route. 

it's okay to leave it as this but might want to consider refactoring this if you have time.

http://callbackhell.com/

- moduli-zing some of the callbacks may solve this

Reply:
Got it, will do!